libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -121,11 +141,25 @@ bool HTTPClient::begin(String url, String httpsFingerprint)
 bool HTTPClient::begin(String url)
 {
     _transportTraits.reset(nullptr);
-    _port = 80;
-    if (!beginInternal(url, "http")) {
+    if (!beginInternal(url, nullptr)) {
+        return false;
+    }
+    if (_protocol == "https" || _port == 443) {
+        _port = 443;
+        _transportTraits = TransportTraitsPtr(new TLSTraits());
+    }
+    else {
+        _port = 80;
+        _transportTraits = TransportTraitsPtr(new TransportTraits());
+    }
+    _tcp = _transportTraits->create();
+
+    if(!_tcp->connect(_host.c_str(), _port)) {
```

Comment:
Agreed. What I'm trying to do is make it a temporary cost; you only need it in RAM during the verification phase, and then you can unload it from RAM. Something like this:
`
    SPIFFS.begin();
    File ca = SPIFFS.open(F("/Equifax.cer"), "r");
    if (ca) {
      if (!firestream->setRootCA(ca, ca.size())) {
        Serial.println(F("Root CA failed to load"));
      }
      ca.close();
    }
    SPIFFS.end();
`